src/network/taraxa_capability.hpp

```diff
@@ -103,68 +103,31 @@ class TaraxaPeer : public boost::noncopyable {
-        db_(db),
-        pbft_mgr_(pbft_mgr),
-        pbft_chain_(pbft_chain),
-        vote_mgr_(vote_mgr),
-        next_votes_mgr_(next_votes_mgr),
-        dag_mgr_(dag_mgr),
-        dag_blk_mgr_(dag_blk_mgr),
-        trx_mgr_(trx_mgr),
-        lambda_ms_min_(lambda_ms_min) {
-    LOG_OBJECTS_CREATE("TARCAP");
-    LOG_OBJECTS_CREATE_SUB("PBFTSYNC", pbft_sync);
-    LOG_OBJECTS_CREATE_SUB("DAGSYNC", dag_sync);
-    LOG_OBJECTS_CREATE_SUB("NEXTVOTESSYNC", next_votes_sync);
-    LOG_OBJECTS_CREATE_SUB("DAGPRP", dag_prp);
-    LOG_OBJECTS_CREATE_SUB("TRXPRP", trx_prp);
-    LOG_OBJECTS_CREATE_SUB("PBFTPRP", pbft_prp);
-    LOG_OBJECTS_CREATE_SUB("VOTEPRP", vote_prp);
-    LOG_OBJECTS_CREATE_SUB("NETPER", net_per);
-    for (uint8_t it = 0; it != PacketCount; it++) {
-      packet_count[it] = 0;
-      packet_size[it] = 0;
-      unique_packet_count[it] = 0;
-    }
-  }
+struct TaraxaCapability : virtual CapabilityFace {
+  TaraxaCapability(Host &_host, ba::io_service &io_service, NetworkConfig const &_conf,
+                   std::shared_ptr<DbStorage> db = {}, std::shared_ptr<PbftManager> pbft_mgr = {},
+                   std::shared_ptr<PbftChain> pbft_chain = {}, std::shared_ptr<VoteManager> vote_mgr = {},
+                   std::shared_ptr<NextVotesForPreviousRound> next_votes_mgr = {},
+                   std::shared_ptr<DagManager> dag_mgr = {}, std::shared_ptr<DagBlockManager> dag_blk_mgr = {},
+                   std::shared_ptr<TransactionManager> trx_mgr = {}, bool performance_log = false,
+                   addr_t const &node_addr = {});
+
   virtual ~TaraxaCapability() = default;
+
   std::string name() const override { return "taraxa"; }
   unsigned version() const override { return 1; }
   unsigned messageCount() const override { return PacketCount; }
 
-  void onStarting() override;
-  void onStopping() override {
-    stopped_ = true;
-    if (conf_.network_simulated_delay > 0) io_service_.stop();
-    dag_blk_mgr_ = nullptr;
-    vote_mgr_ = nullptr;
-    dag_mgr_ = nullptr;
-    trx_mgr_ = nullptr;
-  }
+  bool pbft_syncing() const { return syncing_.load(); }
 
   void onConnect(NodeID const &_nodeID, u256 const &) override;
   void syncPeerPbft(NodeID const &_nodeID, unsigned long height_to_sync);
   void restartSyncingPbft(bool force = false);
   void delayedPbftSync(NodeID _nodeID, int counter);
   std::pair<bool, blk_hash_t> checkDagBlockValidation(DagBlock const &block);
-  bool interpretCapabilityPacket(NodeID const &_nodeID, unsigned _id, RLP const &_r) override;
-  bool interpretCapabilityPacketImpl(NodeID const &_nodeID, unsigned _id, RLP const &_r);
+  void interpretCapabilityPacket(NodeID const &_nodeID, unsigned _id, RLP const &_r) override;
+  void interpretCapabilityPacket_(NodeID const &_nodeID, unsigned _id, RLP const &_r);
+  void interpretCapabilityPacketImpl(NodeID const &_nodeID, unsigned _id, RLP const &_r);
   void onDisconnect(NodeID const &_nodeID) override;
```

Comment:
Can we refactor these names to something meaningful ? Functions/variables names should be as self explanatory as possible. I would have absolutely no clue what might be difference between these 3 methods when named interpretCapabilityPacket, interpretCapabilityPacket_  and interpretCapabilityPacketImpl...

Reply:
This is a demand to refactor the old code